Allow 'mmctl channel move' to move private channels

Leats opened the pull request:
#### Summary
<!--
A description of what this pull request does.
-->
Currently `mmctl channel move someotherteam teamname:channelname` does not allow moving private channels. The error message only mentions direct message and group message channels that cannot be moved:
https://github.com/mattermost/mattermost-server/blob/15037a510a5da29bf16badb5ae4e6c69ee6fe946/i18n/en.json#L339-L340
While not being able to move direct message and group message channels makes sense, the same should not apply to private channels.

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->
None

#### Release Note
<!--
If no release notes are required, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your past-tense release note in the block below.
-->
```release-note
Fixed an issue where `mmctl channel move` did not allow moving private channels.
```

mattermod commented:
Hello @Leats,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

ThiefMaster reviewed (approved):
Very useful fix! I had to make a channel temporarily public. move it, and then make it private again as a workaround (ugly, because in theory this could leak data if someone managed to join just in that moment), and the old `mattermost channel move ...` command did not have this problem either.

streamer45 reviewed (approved):
Thank you @Leats.

Changes look good to me. I am just not sure if this was an overlook or if there was a reason to limit this to public channels only. Will defer to @mgdelacroix on that.

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @jfrerich @emilyacook

jfrerich commented:
@mgdelacroix, ping for review (and welcome back from the holiday break)

mgdelacroix reviewed (commented):
Thanks a lot for the contribution, @Leats! The change LGTM, I would only like to increase the test coverage.

Would you be open to add test cases for `Should fail to move channel due to a member not member of target team` and `Should be able to (force) move channel by a member that is not member of target team` with a private channel? Both exist already in the same test function for public channels.

Leats commented:
> Would you be open to add test cases for `Should fail to move channel due to a member not member of target team` and `Should be able to (force) move channel by a member that is not member of target team` with a private channel?

Thank you! Yes, I can add that as well! ☺️

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @jfrerich @emilyacook

emilyacook commented:
> This PR has been automatically labelled "stale" because it hasn't had recent activity.
> A core team member will check in on the status of the PR to help with questions.
> Thank you for your contribution!
> 

Hey @Leats! Let me know if there is any update on this :)

Leats commented:
Hello @emilyacook @mgdelacroix! 
Very sorry for the wait, my move took longer than expected. I added the tests as requested, I hope everything is alright! :)

mgdelacroix commented on api4/channel_test.go:
```diff
@@ -4016,7 +4016,22 @@ func TestMoveChannel(t *testing.T) {
+		_, resp = client.AddChannelMember(privateChannel.Id, user.Id)
+		CheckNoError(t, resp)
+
+		_, resp = th.SystemAdminClient.MoveChannel(privateChannel.Id, team2.Id, false)
```
```suggestion
		_, resp = client.MoveChannel(privateChannel.Id, team2.Id, false)
```

mgdelacroix commented on api4/channel_test.go:
```diff
@@ -4031,5 +4046,20 @@ func TestMoveChannel(t *testing.T) {
+		_, resp = client.AddChannelMember(privateChannel.Id, user.Id)
+		CheckNoError(t, resp)
+
+		newChannel, resp := th.SystemAdminClient.MoveChannel(privateChannel.Id, team2.Id, true)
```
```suggestion
		newChannel, resp := client.MoveChannel(privateChannel.Id, team2.Id, true)
```

mgdelacroix reviewed (changes requested):
Thanks for adding the tests @Leats! Added a couple of small changes

Leats commented:
Thank you, @mgdelacroix! I changed the code as requested and added the same behaviour to the local channel move so it's consistent with my previous changes (and lets the tests run through).

mgdelacroix reviewed (approved):
Thanks a lot @Leats! LGTM!! :+1:

lindy65 commented:
Hi @mgdelacroix - Can you link me to the Jira ticket for this PR please? Thanks :)

ThiefMaster commented:
I don't think there is one for it - it's just an annoying issue we discovered on our Mattermost instance when moving channels for a user, so I suggested @Leats that sending a PR for this would be a nice contribution.

lindy65 commented:
Thanks @ThiefMaster - QA requires a Jira ticket to move the PR through the process so wondering whether you'd like to create one? :)

ThiefMaster commented:
I don't think people who aren't mattermost staff can do that, can they?

> You do not have permission to create issues in this project.

lindy65 commented:
Ah, you're correct @ThiefMaster - non-staff can view Jira tickets only.  So @mgdelacroix will need to help us out here :)

lindy65 commented:
Hi @mgdelacroix :) I've created a [jira ticket](https://mattermost.atlassian.net/browse/MM-32746) for this PR. Please help fill in QA test steps and description?  TIA 👍

mgdelacroix commented:
Thanks @lindy65!! done :)

lindy65 reviewed (approved):
Thanks @Leats - looks good to me.  Testing will be done after merge 👍
